[Pebble] Http function should log the error properly

### Describe the issue

When using the HTTP pebble function, if the answer is not correct, we just say that we couldnt perform the request :
`2025-07-31 14:06:06.585io.pebbletemplates.pebble.error.PebbleException: Unable to execute HTTP request ({{ http(uri='https://jsonplaceholder.typicode.com/post', method='GET', body=inputs.json) }}:1)`

But reading the trace, we did execute the request, but it returned an error:
`Caused by: io.kestra.core.http.client.HttpClientResponseException: Failed http request with response code '404' and body:[0x36][0x37]�[0x39][0x3a][0x3b][0x3c][0x3d][0x3e][0x3f]���[0x43][0x44]C���[0x49][0x4a][0x4b][0x4c]`

### Environment

- Kestra Version: develop

--- a/core/src/main/java/io/kestra/core/runners/pebble/functions/HttpFunction.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/functions/HttpFunction.java
@@ -5,6 +5,8 @@ import io.kestra.core.http.HttpRequest;
 import io.kestra.core.http.HttpResponse;
 import io.kestra.core.http.client.HttpClient;
 import io.kestra.core.http.client.HttpClientException;
+import io.kestra.core.http.client.HttpClientRequestException;
+import io.kestra.core.http.client.HttpClientResponseException;
 import io.kestra.core.http.client.configurations.HttpConfiguration;
 import io.kestra.core.runners.RunContext;
 import io.kestra.core.runners.RunContextFactory;
@@ -101,8 +103,15 @@ public class HttpFunction<T> implements Function {
         try (HttpClient httpClient = new HttpClient(runContext, httpConfiguration)) {
             HttpResponse<Object> response = httpClient.request(httpRequest, Object.class);
             return response.getBody();
-        } catch (HttpClientException | IllegalVariableEvaluationException | IOException e) {
-            throw new PebbleException(e, "Unable to execute HTTP request", lineNumber, self.getName());
+        } catch (HttpClientResponseException e) {
+            if (e.getResponse() != null) {
+                String msg = "Failed to execute HTTP Request, server respond with status " + e.getResponse().getStatus().getCode() + " : " + e.getResponse().getStatus().getReason();
+                throw new PebbleException(e, msg , lineNumber, self.getName());
+            } else {
+                throw new PebbleException( e, "Failed to execute HTTP request ", lineNumber, self.getName());
+            }
+        } catch(HttpClientException | IllegalVariableEvaluationException | IOException e ) {
+            throw new PebbleException( e, "Failed to execute HTTP request ", lineNumber, self.getName());
         }
     }
 

--- a/core/src/test/java/io/kestra/core/runners/pebble/functions/HttpFunctionTest.java
+++ b/core/src/test/java/io/kestra/core/runners/pebble/functions/HttpFunctionTest.java
@@ -8,6 +8,7 @@ import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import io.kestra.core.runners.VariableRenderer;
 import io.kestra.core.junit.annotations.KestraTest;
 import io.kestra.core.serializers.JacksonMapper;
+import io.pebbletemplates.pebble.error.PebbleException;
 import jakarta.inject.Inject;
 import org.apache.hc.client5.http.utils.Base64;
 import org.junit.jupiter.api.Assertions;
@@ -19,6 +20,8 @@ import java.util.Map;
 
 import static com.github.tomakehurst.wiremock.client.WireMock.*;
 import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertThrows;
 
 @KestraTest
 @WireMockTest(httpPort = 28182)
@@ -43,6 +46,13 @@ class HttpFunctionTest {
             ))
         );
         Assertions.assertTrue(rendered.contains("\"todo\":\"New todo\""));
+    }
+
+    @Test
+    void wrongMethod() {
+        var exception = assertThrows(IllegalVariableEvaluationException.class, () -> variableRenderer.render("{{ http(url) }}", Map.of("url", "https://dummyjson.com/todos/add")));
+        assertThat(exception.getCause()).isInstanceOf(PebbleException.class);
+        assertThat(exception.getCause().getMessage()).isEqualTo("Failed to execute HTTP Request, server respond with status 404 : Not Found ({{ http(url) }}:1)");
     }
 
     @Test

